server/src/session_manager.ts

```diff
@@ -98,32 +151,74 @@ export class SessionManager {
         globalThis.window = dom.window as any;
         globalThis.document = dom.window.document;
 
-        const bgConfig = {
-            fetch: (url: any, options: any) => fetch(url, options),
+        let dispatcher: Agent | undefined;
+        let proxies: any;
+        if (proxy) {
+            dispatcher = this.getProxyDispatcher(proxy);
+        } else {
+            proxies = {
+                http: process.env.HTTP_PROXY,
+                https: process.env.HTTPS_PROXY || process.env.HTTP_PROXY,
+                ftp: process.env.FTP_PROXY,
+            };
+        }
+
+        const bgConfig: BgConfig = {
+            fetch: async (url: any, options: any): Promise<any> => {
+                if (proxies) {
+                    const parsedUrl = new URL(url);
+                    dispatcher = this.getProxyDispatcher(
```

Comment:
Isn't the protocol always going to be http?

Reply:
actually it's always https.
it's just following the standards

Reply:
if its always https, can we just replace this with `this.getProxyDispatcher(proxies['http'])`?

Reply:
you can even just do this
```ts
dispatcher = this.getProxyDispatcher(process.env.HTTPS_PROXY || process.env.HTTP_PROXY)
```